src/sentry/integrations/api/serializers/rest_framework/data_forwarder.py

```diff
@@ -0,0 +1,263 @@
+        aws_region_pattern = r"^[a-z0-9\-]+$"
+        region = config.get("region")
+        if not region or not re.match(aws_region_pattern, region):
+            errors.append("region must be a valid AWS region format")
+
+    def _validate_sqs_credentials(self, config: dict, errors: list[str]) -> None:
+        access_key = config.get("access_key")
+        secret_key = config.get("secret_key")
+
+        if not access_key or access_key.strip() == "":
+            errors.append("access_key must be a non-empty string")
+
+        if not secret_key or secret_key.strip() == "":
+            errors.append("secret_key must be a non-empty string")
+
+    def _validate_sqs_message_group_id(self, config: dict, errors: list[str]) -> None:
+        message_group_id = config.get("message_group_id")
+        queue_url = config.get("queue_url")
+
+        if message_group_id is not None and message_group_id.strip() == "":
+            errors.append("message_group_id must be a non-empty string or null")
+
+        if isinstance(queue_url, str) and queue_url.endswith(".fifo") and not message_group_id:
+            errors.append("message_group_id is required for FIFO queues")
+
+    def _validate_sqs_s3_bucket(self, config: dict, errors: list[str]) -> None:
+        s3_bucket = config.get("s3_bucket")
+        if s3_bucket is not None:
+            if not s3_bucket or s3_bucket.strip() == "":
+                errors.append("s3_bucket must be a non-empty string")
+            else:
+                s3_bucket_pattern = r"^[a-z0-9\-\.]+$"
+                if not re.match(s3_bucket_pattern, s3_bucket):
+                    errors.append("s3_bucket must be a valid S3 bucket name")
+
+    def _validate_segment_write_key(self, config: dict) -> None:
+        segment_write_key_pattern = r"^[a-zA-Z0-9_\-]+$"
+        write_key = config.get("write_key")
+        if not write_key or not re.match(segment_write_key_pattern, write_key):
+            raise ValidationError("write_key must be a valid Segment write key format")
+
+    def _validate_splunk_instance_url(self, config: dict, errors: list[str]) -> None:
+        splunk_url_pattern = r"^https?://[a-zA-Z0-9\-\.]+(?::\d+)?(?:/.*)?$"
+        instance_url = config.get("instance_URL")
+        if not instance_url or not re.match(splunk_url_pattern, instance_url):
+            errors.append("instance_URL must be a valid URL starting with http:// or https://")
+
+    def _validate_splunk_required_strings(self, config: dict, errors: list[str]) -> None:
+        index = config.get("index")
+        if not index or index.strip() == "":
+            errors.append("index must be a non-empty string")
+
+        source = config.get("source")
+        if not source or source.strip() == "":
+            errors.append("source must be a non-empty string")
+
+        token = config.get("token")
+        if not token or token.strip() == "":
+            errors.append("token must be a non-empty string")
```

Comment:
I think if we set allow_blank to be false in the CharField serializer above then we should be able to reduce down these blank string checks